### ISSUE TITLE
Render `access=` on `highway=pedestrian`?

### ISSUE THREAD

**Issue by osmuser63783** [gravitystorm/openstreetmap-carto#4998]
### Expected behavior

Show `highway=pedestrian access=no` with a pattern similar to the one used for other streets with `access=no`

### Actual behavior
No rendering of `access=` on `highway=pedestrian`. I would be curious if this is a deliberate decision - as it it seems inconsistent with the rendering of access restrictions in other highway types.

I noticed this when looking at a pedestrian bridge that had been closed and marked with `access=no` (it has since reopened). I was surprised to see that it was tagged `access=no` but did not "look closed" on the map. 

Searching Overpass for examples of `access=no` on `highway=pedestrian` (4,395 cases worldwide) finds more examples of this e.g. a pedestrian street being closed (to pedestrians) due to roadworks for a couple of months. It also shows examples of verbose access tagging, e.g. `highway=pedestrian access=no foot=yes bicycle=yes motor_vehicle=no`, so rendering `access` on `pedestrian` without taking into account `foot` (i.e. solving #214) would have the side effect of showing these on the map. Not sure if that would be considered a good or a bad thing, but I wanted to flag it.

### Screenshots with links illustrating the problem
![Screenshot_20240802_002643_Firefox](https://github.com/user-attachments/assets/2b8b60eb-6d36-4340-bffc-ebfcc31cad8d)
This is a screenshot of https://www.openstreetmap.org/way/5049168 before I removed `access=no` (because the bridge has now reopened).

**Comment by imagico** [comment]
Yes, this is one of the things we discussed as potential follow-up changes to #4952.  With the current system of showing only `access=*` and not the transport mode specific access tags showing `access=no` without interpreting `foot=*` would often be confusing.  But with #4952 this could definitely be of value.

**Comment by dch0ph** [comment]
The straightforward fix would be to add rendering of `restricted` and `no` access marking for `highway=pedestrian` based on tag interpretation for footways already added in #4952 . 

The only decision would be the colour of access marking to provide sufficient contrast with the grey colour of `highway=pedestrian`. I would suggest simply white.

A PR could include #5024 , since the logic is the same?
